run_model.py

```diff
@@ -18,67 +21,186 @@
+    configs_to_analyze = []
+    for analysis_field in analysis_config:
+        for usage_field in server_config["usage"]:
+            if usage_field == analysis_field:
+                model_params = generate_model_params(server_config["usage"][usage_field], analysis_config[analysis_field], usage_field)
+                # Generate a different cluster config for each entry in model_params
+                for model_param in model_params:
+                    analysis_model_dict = copy.deepcopy(server_config)
+                    analysis_model_dict["usage"][usage_field] = model_param['model']
+                    # Write the config to a temp file so it can be make into a ClusterConfig object
+                    with open(tempfile.NamedTemporaryFile().name, 'a+') as f:
+                        f.write(json.dumps(analysis_model_dict))
+                        f.seek(0)
+                        analysis_model_config = cluster_config_from_path(f.name)
+                    configs_to_analyze.append({"model": analysis_model_config,
+                                               "field_to_modify": model_param["field_to_modify"],
+                                               "subfield_to_modify": model_param["subfield_to_modify"],
+                                               "step_value": model_param["step_value"]})
+
+    return configs_to_analyze
+
+
+def generate_model_params(model_params, analysis_config_entry, field_to_modify):
+    step_values = generate_steps(analysis_config_entry)
+    analysis_model_params = []
+    subfield_to_modify = analysis_config_entry["field_name"]
+    for step_value in step_values:
+        this_model_def = copy.deepcopy(model_params)
+        this_model_def[subfield_to_modify] = step_value * model_params[subfield_to_modify]
+        analysis_model_params.append({"model": this_model_def,
+                                      "field_to_modify": field_to_modify,
+                                      "subfield_to_modify": subfield_to_modify,
+                                      "step_value": step_value})
+    return analysis_model_params
+
+def generate_steps(analysis_config_entry):
+    val_to_add = analysis_config_entry["factor_start"]
+    step_values = []
+    while val_to_add <= analysis_config_entry["factor_end"]:
+        step_values.append(val_to_add)
+        val_to_add += analysis_config_entry["factor_step"]
+    return step_values
+
+def compare_to_baseline_values(field_to_update, output_data, baseline_step_value_index):
+    return ['{}%'.format(value/(output_data[field_to_update][baseline_step_value_index]) * 100) for value in output_data[field_to_update]]
 
 if __name__ == '__main__':
+
     parser = argparse.ArgumentParser('CommCare Cluster Model')
-    parser.add_argument('config', help='Path to config file')
+    parser.add_argument('server_config', help='Path to cluster config file')
+    parser.add_argument('-a', '--analysis_config', help='Path to analysis config file', required=False, )
```

Comment:
I think it would make more sense to have a completely separate entry point for the analysis instead of branching the logic in this file at the highest level.

Reply:
https://github.com/dimagi/commcare_resource_model/pull/25/commits/c532b509597587372b868bc8afdea79c2b67e60b